src/sql/workbench/services/query/common/gridDataProvider.ts

```diff
@@ -45,94 +49,144 @@ export interface IGridDataProvider {
 	readonly canSerialize: boolean;
 
 	serializeResults(format: SaveFormat, selection: Slick.Range[]): Thenable<void>;
-
 }
 
-export async function getResultsString(provider: IGridDataProvider, selection: Slick.Range[], includeHeaders?: boolean, tableView?: IDisposableDataProvider<Slick.SlickData>): Promise<string> {
-	let headers: Map<number, string> = new Map(); // Maps a column index -> header
-	let rows: Map<number, Map<number, string>> = new Map(); // Maps row index -> column index -> actual row value
-	const eol = provider.getEolString();
-
-	// create a mapping of the ranges to get promises
-	let tasks: (() => Promise<void>)[] = selection.map((range) => {
-		return async (): Promise<void> => {
-			let startCol = range.fromCell;
-			let startRow = range.fromRow;
-			let result;
-			if (tableView && tableView.isDataInMemory) {
-				// If the data is sorted/filtered in memory, we need to get the data that is currently being displayed
-				const tableData = await tableView.getRangeAsync(range.fromRow, range.toRow - range.fromRow + 1);
-				result = tableData.map(item => Object.keys(item).map(key => item[key]));
-			} else {
-				result = (await provider.getRowData(range.fromRow, range.toRow - range.fromRow + 1)).rows;
-			}
-			// If there was a previous selection separate it with a line break. Currently
-			// when there are multiple selections they are never on the same line
-			let columnHeaders = provider.getColumnHeaders(range);
-			if (columnHeaders !== undefined) {
-				let idx = 0;
-				for (let header of columnHeaders) {
-					headers.set(startCol + idx, header);
-					idx++;
-				}
-			}
-			// Iterate over the rows to paste into the copy string
-			for (let rowIndex: number = 0; rowIndex < result.length; rowIndex++) {
-				let row = result[rowIndex];
-				let cellObjects = row.slice(range.fromCell, (range.toCell + 1));
-				// Remove newlines if requested
-				let cells = provider.shouldRemoveNewLines()
-					? cellObjects.map(x => removeNewLines(x.displayValue))
-					: cellObjects.map(x => x.displayValue);
-
-				let idx = 0;
-				for (let cell of cells) {
-					let map = rows.get(rowIndex + startRow);
-					if (!map) {
-						map = new Map();
-						rows.set(rowIndex + startRow, map);
-					}
+interface Range {
```

Comment:
this is already defined elsewhere.  I guess it doesn't make sense to add the export vscode edit there for such a simple interface.

Reply:
the existing similar interface is the Slick.Range, which is a 2 dimensional range, what I have here is a one dimensional, this is local to the file, I didn't export it.